src/App.tsx

```diff
@@ -1,14 +1,31 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
 import React from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => (
+    this.setState({ query: event.target.value }));
 
   render() {
+    const { query } = this.state;
+
+    const visibleMovies = moviesFromServer.filter(
+      movie => (
+        movie.title.toLowerCase().includes(query.toLocaleLowerCase())
+        || movie.description.toLowerCase().includes(query.toLocaleLowerCase())
+      ),
+    );
```

Comment:
Take this logic to a separate method and call it in the render method. For example: `getVisibleMovies()`

Reply:
Agree, good advice.